### PR TITLE
Add 'displayName' to Template

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
Fixes: https://github.com/openshift/origin/issues/2179

@smarterclayton api change review please?

**Comment by liggitt** [comment]
@jwforres @spadgett for UI to take advantage of when present

**Comment by mfojtik** [comment]
@liggitt @smarterclayton I wonder whether it won't be cleaner to add `Annotation` field into Parameter to make it consistent with other objects and then move the `description` to become Annotation. We can use Annotation(s) for other things, like suggestions on how to use the parameter, etc.

**Comment by liggitt** [comment]
hmm... that's a lot of nesting... I feel like id (name), display name, description are things we'd want first-class support for

**Comment by mfojtik** [comment]
@liggitt sure, I'm fine with adding `displayName` as first class citizen... Just feel weird that we do that only for Parameter (AFAIK).

**Comment by smarterclayton** [comment]
API change is ok

**Comment by smarterclayton** [comment]
As long as display name continues to be optional

**Comment by mfojtik** [comment]
@smarterclayton it should stay optional, we fallback to `name` if `displayName` is empty (UI should be aware of that).

**Comment by mfojtik** [comment]
@smarterclayton @liggitt is this OK to merge?

File: pkg/cmd/cli/describe/describer.go [hunk 1]
```diff
@@ -671,8 +671,15 @@ func (d *TemplateDescriber) DescribeParameters(params []templateapi.Parameter, o
 	formatString(out, "Parameters", " ")
 	indent := "    "
 	for _, p := range params {
-		formatString(out, indent+"Name", p.Name)
-		formatString(out, indent+"Description", p.Description)
+		if len(p.DisplayName) > 0 {
+			formatString(out, indent+"Name", p.DisplayName)
+			formatString(out, indent+"Parameter Name", p.Name)
```
Comment (liggitt) [inline pkg/cmd/cli/describe/describer.go]: don't change the label for `Name`, just call display name `Display Name`

**Comment by liggitt** [comment]
nit on describer, then lgtm

**Comment by mfojtik** [comment]
@liggitt you're right. fixed.

[merge]

**Comment by mfojtik** [comment]
re [merge]

**Pull request merged**